src/pip/_internal/commands/install.py

```diff
@@ -221,11 +221,12 @@ def run(self, options, args):
 
         global_options = options.global_options or []
 
-        with self._build_session(options) as session:
+        with self._build_session(options) as session, \
+                WheelCache(options.cache_dir, options.format_control) as  \
+                wheel_cache:
```

Comment:
This looks super ugly but it is the best that I could come up with. If anyone else has any better ideas, I am open to them.

Reply:
Why not put a cleanup call in the finally of the try clause starting below at line 261 or 251?

I don't see why this needs to be a context manager. It doesn't make sense to me.

Reply:
It needs to be a context manager because the files were not cleaned up correctly before, as you pointed out.

Reply:
> needs to be a context manager

I think just calling the cleanup method in the existing finally clause should be just fine.

I don't want to add one-more-way to use a `WheelCache`, especially since this behaviour is not really a thing in `Cache`.